### ISSUE TITLE
Migrate /users/sso functionality to API v2

### ISSUE THREAD

**Issue by initvector** [vanilla/vanilla#5770]
The [/users/sso](https://blog.vanillaforums.com/api-users-sso) endpoint from API v1 (or similar functionality) needs to be migrated to API v2.

Depends on #5756

**Comment by initvector** [comment]
On hold pending technical direction from @tburry.

**Comment by linc** [comment]
1. Decision needs to be made about SmartIDs in API v2 before we work on this.
2. Needs a full spec as a separate task.
